tsconfig.json

```diff
@@ -10,15 +10,17 @@
     // "allowJs": true /* Allow javascript files to be compiled. */,
     // "checkJs": true,                       /* Report errors in .js files. */
     "jsx": "react" /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */,
-    // "declaration": true /* Generates corresponding '.d.ts' file. */,
-    // "declarationMap": true /* Generates a sourcemap for each corresponding '.d.ts' file. */,
+    "declaration": true /* Generates corresponding '.d.ts' file. */,
+    "declarationMap": true /* Generates a sourcemap for each corresponding '.d.ts' file. */,
```

Comment:
question: is this useful? It pops a sourcemap into the `types/` directory but I'm not sure anyone would use it?

Reply:
Does VS Code pick this up when I hit Go To Definition? That's what I would assume to use it for.

I'm inclined to thing they're not useful, though.

Reply:
another question: Do we still need the `.d.ts` files? Can't we just distribute `src/` now?

Reply:
Not the individual ones, just the main `index.d.ts`. Rewinding back a bit, I would expect that to be in the package root like it is now.

We already put the `src` into the package verbatim.

Reply:
so is there any reason we can't do `"typings": "src/index.ts"`?

Reply:
> Not the individual ones, just the main `index.d.ts`. Rewinding back a bit, I would expect that to be in the package root like it is now.

I tried to do this, it isn't possible in an automatic way, because typescript automatically ignores the directory we tell it to pop its results into, probably to avoid infinite loops.

We can just manually `cp types/index.d.ts .` but it seems unnecessary to me. I haven't seen any other examples out there that keep any autogenerated files in the repo

Also, I checked, and no one is using `"types": "index.ts"` so I'm going to assume it's a bad idea

Reply:
Ah yes, we don't need definitions because we are fully TS'ed now.

Reply:
@timdorr "Go to Definition" doesn't work like that with sourcemaps.

@cellog `"types": "index.ts"` is definitely a bad idea, as the types are supposed to point to a `.d.ts` file. All TS files should be gone when you publish to npm.

@timdorr do you mean we don't need manually-typed definitions? Correct. Still need auto-generated ones though.

Reply:
We publish `src` now. We want that in the package so the source is always available to read through. It's small and I consider it an important feature of Redux.

Whether we use that for our types or not, it's going to stay in the package.